query_optimizer/resolver/Resolver.cpp

```diff
@@ -2656,11 +2666,12 @@ E::ScalarPtr Resolver::resolveFunctionCall(
   }
 
   // Make sure a naked COUNT() with no arguments wasn't specified.
-  if ((aggregate->getAggregationID() == AggregationID::kCount)
-      && (resolved_arguments.empty())
-      && (!count_star)) {
-    THROW_SQL_ERROR_AT(&parse_function_call)
-        << "COUNT aggregate requires an argument (either scalar or star (*))";
+  if ((aggregate != nullptr && aggregate->getAggregationID() == AggregationID::kCount)
```

Comment:
Code style for the bool expressions:

https://google.github.io/styleguide/cppguide.html#Boolean_Expressions